src/biokbase/narrative/jobs/job.py

```diff
@@ -240,6 +240,28 @@ def was_terminal(self):
         else:
             return self._acc_state.get("status") in TERMINAL_STATUSES
 
+    @property
+    def is_terminal(self):
+        self.state()
+        if self._acc_state.get("batch_job"):
+            for child_job in self.children:
+                if child_job._acc_state.get("status") != COMPLETED_STATUS:
+                    child_job.state(force_refresh=True)
+        return self.was_terminal
+
+    def in_cells(self, cell_ids):
```

Comment:
Could you add type hints to this function?

What happens if `cell_ids` is `[]` or `None`?

Reply:
It should be able to handle `[]` intuitively.

TBH error handling is a bit beyond me in this program. My instinct is to just let the natural error of `None` not being iterable to happen so the programmer can see the error and debug

Reply:
How about throwing a simple `ValueError` if `cell_id`s is None so it's easier to debug than the iterable error text?